pykokkos/core/translators/bindings.py

```diff
@@ -138,44 +140,45 @@ def generate_functor_instance(functor: str, members: PyKokkosMembers) -> str:
     for f in members.fields:
         args.append(f.declname)
 
-    exec_space: str = Keywords.DefaultExecSpace.value
-    exec_space_instance: str = Keywords.DefaultExecSpaceInstance.value
+    if functorExecSpace is None:
+        exec_space: str = Keywords.DefaultExecSpace.value
+        exec_space_instance: str = Keywords.DefaultExecSpaceInstance.value
+    else:
+        exec_space: str = functorExecSpace
+        exec_space_instance: str = f"{exec_space}_instance"
+
     mirror_views: str = f"{exec_space} {exec_space_instance};"
 
     device_views: Dict[str, str] = get_device_views(members)
     for v, d_v in device_views.items():
         args.append(d_v)
-
-        view_type: cppast.ClassType = members.views[cppast.DeclRefExpr(v)]
-        if get_view_memory_space(view_type, "bindings") == Keywords.ArgMemSpace.value:
-            mirror_views += f"auto {d_v} = Kokkos::create_mirror_view_and_copy({exec_space_instance}, {v});"
-        else:
-            mirror_views += f"auto {d_v} = {v};"
+        mirror_views += f"auto {d_v} = Kokkos::create_mirror_view_and_copy({exec_space_instance}, {v});"
```

Comment:
What is the reason for this change? This helps avoid calling `create_mirror_view_and_copy` when the data is already in the right place, I believe it's used when passing Unmanaged views and for UVM space. Is it a no-op if the data is already on the device?

Reply:
so `create_mirror_view_and_copy` should take care of this, the documentation says:
If `std::is_same<typename Space::memory_space, typename ViewType::memory_space>::value` is false, creates a new `Kokkos::View` with the same layout and padding as `src` but with a device type of `Space::device_type` and conducts a `deep_copy` from `src` to the new view if one was created. Otherwise returns `src`.

I would just leave this to the Kokkos Core lib as adding another logic outside could break if anything changes within Kokkos. This is basically the reason we have these functions in the first place.
But apart from this: The cast operator uses this function to generate the constructor call. If it is dependent of the memory space, we would need a different version of the cast function for the case of the same memory space in `src` and `dst`.  That is why it made sense to do it in this pr (given that the kokkos function does the same logic we were repeating).

Reply:
I talked to @NaderAlAwar about this and we decided to introduce a selector parameter which switches the behavior from the old one to a new version that always uses `create_mirror_view_and_copy`. [70f167](https://github.com/kokkos/pykokkos/pull/107/commits/70f167dd3f1747e61561150d94370bd5da041b9a)